src/extractor/guidance/intersection_handler.cpp

```diff
@@ -38,13 +38,17 @@ IntersectionHandler::IntersectionHandler(const util::NodeBasedDynamicGraph &node
 {
 }
 
-std::size_t IntersectionHandler::countValid(const Intersection &intersection) const
+std::size_t IntersectionHandler::countValid(const ConnectedRoads &intersection) const
 {
     return std::count_if(intersection.begin(), intersection.end(), [](const ConnectedRoad &road) {
         return road.entry_allowed;
     });
 }
 
+// Checks whether a turn from `via_edge` onto `road` is a turn
+// - on a ramp
+// - a continue
+// - or a normal turn
```

Comment:
What does this mean wrt. our turn instructions?

https://github.com/Project-OSRM/osrm-backend/blob/master/include/extractor/guidance/turn_instruction.hpp#L33